cranelift-codegen/src/isa/x86/enc_tables.rs

```diff
@@ -405,12 +405,21 @@ fn expand_fcvt_from_uint(
     let mut pos = FuncCursor::new(func).at_inst(inst);
     pos.use_srcloc(inst);
 
-    // Conversion from unsigned 32-bit is easy on x86-64.
-    // TODO: This should be guarded by an ISA check.
-    if xty == ir::types::I32 {
-        let wide = pos.ins().uextend(ir::types::I64, x);
-        pos.func.dfg.replace(inst).fcvt_from_sint(ty, wide);
-        return;
+    // Conversion from an unsigned int smaller than 64bit is easy on x86-64.
+    match xty {
+        ir::types::I8 | ir::types::I16 => {
+            let wide = pos.ins().uextend(ir::types::I64, x);
+            pos.func.dfg.replace(inst).fcvt_from_sint(ty, wide);
+            return;
+        }
+        ir::types::I32 => {
+            // TODO: This should be guarded by an ISA check.
```

Comment:
This comment applies to the `I8` and `I16` cases too, and then those and the `I32` cases are the same code, so we can merge them into one match arm.

Reply:
I dont think that comment applies to `I8` and I16`, as those have to be `uextend`ed on 32bit x86 too. Though in that case they can be extended to `I32`. I wanted to merge both arms, but wasnt sure how to change the comment to make clear it only applies to `I32`.

Reply:
We'll still want an ISA check for `I8` and `I16`, because we can't expand them to  `I64` operations on x86-32.